src/main/scala/org/allenai/plugins/Utilities.scala

```diff
@@ -49,4 +49,25 @@ object Utilities {
     }
     IO.write(destination, contents)
   }
+
+  /** Given a set of files on disk, produce a consistent SHA1 hash of their contents. This hash will
+    * change when the contents of the files change, or when the file name changes relative to the
+    * given file.
+    * @param filesToHash the files whose contents and names should be hashed
+    * @param rootDir the root directory the file names should be resolved against before hashing.
+    * Only path changes relative to this directory will cause the hash to change.
+    */
+  def hashFiles(filesToHash: Seq[File], rootDir: File): String = {
+    // Resolve the filenames relative to the root directory.
+    val rootDirPath = rootDir.toPath
+    val relativizedNames = filesToHash.map(_.toPath).map(rootDirPath.relativize).map(_.toString)
```

Comment:
I assume relativize will create a canonical path?  If I feed it "foo/dir/../dir" what happens?  Probably add a test for this.